src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/Documents/VisualStudioFileChangeTracker.cs

```diff
@@ -78,7 +84,7 @@ public override void StartListening()
                 return;
             }
 
-            _fileChangeAdviseTask = _joinableTaskFactory.RunAsync(async () =>
+            _fileChangeAdviseTask = _joinableTaskContext.Factory.RunAsync(async () =>
```

Comment:
Oh wow, the `_fileChangeService` is free-threaded and doesn't require the UI thread?

Reply:
It seems like it! I didn't run into any trouble with it not running on the UI thread